generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -164,6 +166,175 @@ public OpaqueExpressionGenerator(IRandom generator, GenerationParams generationP
+    } else {
+      assert !isZero;
+      assert operator == BinOp.SHL || operator == BinOp.SHR;
+      // We're going to shift twice in opposite directions by the same value.
+      final BinOp secondOp = operator == BinOp.SHL ? BinOp.SHR : BinOp.SHL;
+      return Optional.of(
+          new ParenExpr(
+              new BinaryExpr(
+                  new ParenExpr(
+                      new BinaryExpr(
+                          makeOpaqueOne(type, constContext, depth, fuzzer),
+                          applyIdentityFunction(shiftValue.clone(), type, constContext, depth,
+                              fuzzer),
+                          operator)),
+                  applyIdentityFunction(shiftValue.clone(), type, constContext, depth, fuzzer),
+                  secondOp)));
+    }
+  }
+
+  /**
+   * Function to generate an opaque value by performing a bitwise operation on an opaque zero or
+   * an opaque one.
+   * Possibilities to create zero include:
+   * Bitwise ANDing a fuzzed expression with an opaque zero: (fuzzedexpr) & (opaque zero)
+   * Bitwise ORing an opaque zero with an opaque zero: (opaque zero) | (opaque zero)
+   * Bitwise XORing an opaque zero with an opaque zero or an opaque one with an opaque one:
+   *     (opaque zero) ^ (opaque zero) or (opaque one) ^ (opaque one)
+   * Possibilities to create one include:
+   * Bitwise ANDing an opaque one with an opaque one: (opaque one) & (opaque one)
+   * Bitwise ORing an opaque one with an opaque zero or one: (opaque one) | (opaque zero or one)
+   * Bitwise XORing an opaque zero with an opaque one or an opaque one with an opaque zero:
+   *     (opaque zero) ^ (opaque one) or (opaque one) ^ (opaque zero)
+   *
+   * @param type - the base type of the opaque value being created.
+   * @param constContext - true if we're in a constant expression context, false otherwise.
+   * @param depth - how deep we are in the expression.
+   * @param fuzzer - the fuzzer object for generating fuzzed expressions.
+   * @param isZero - true if we are making an opaque zero, false otherwise.
+   * @return Optional.empty() if an opaque value can't be generated, otherwise an opaque value
+   *     made from performing a bitwise operation on an opaque zero or opaque one.
+   */
+  private Optional<Expr> opaqueZeroOrOneBitwiseOp(BasicType type, boolean constContext,
+                                                   final int depth, Fuzzer fuzzer, boolean isZero) {
+    if (!BasicType.allIntegerTypes().contains(type)) {
+      return Optional.empty();
+    }
+    if (!shadingLanguageVersion.supportedBitwiseOperations()) {
+      return Optional.empty();
+    }
+    final int operator = generator.nextInt(3);
+    Optional<Expr> opaqueExpr;
+    switch (operator) {
+      case 0:
+        // Bitwise AND
+        if (isZero) {
+          opaqueExpr = Optional.of(
+              new ParenExpr(
+                  generator.nextBoolean()
+                  ? new BinaryExpr(
```

Comment:
When passing 'constContext' to 'fuzzExpr', you should pass the literal 'true', because these fuzzed expressions really wil get evaluated (no short-circuit evaluation for bitwise operators).